### DISCUSSION TITLE
Python: "TypeError: ord() expected string of length 1, but int found"

### DISCUSSION THREAD

**Discussion by agnibhu13** [antlr/antlr4#4490] [General]
![ord_Issue](https://github.com/antlr/antlr4/assets/124293916/736c1a9f-97dd-41cb-aeb1-2dff4b261122)

Hi,

I am not able to run antlr4 with Python, getting a ' ord() expected string of length 1, but int found' error. The screenshot is attached.

My setup:
Python 3.12.1
ANTLR Parser Generator  Version 4.13.1

Can someone help me?

**Comment by ericvergnaud** [comment]
Looks like the Python antlr runtime you’re using is not 4.13.1 ?

**Reply by agnibhu13** [reply]
thanks a lot for the comment. Yeah, the Python antlr runtime was not 4.13.1, instead it was (4.9.3).
It was a dependency issue, Another Python module needed the old version so the latest version was not installed. I installed the 4.13.1 and it is working fine now.
